localstack/utils/generic/wait_utils.py

```diff
@@ -0,0 +1,28 @@
+import time
+from typing import Callable
+
+from typing_extensions import Literal
+
+
+def wait_until(
```

Comment:
Nice util function. 👍  At some point we could also move `retry(..)` from `utils/common.py` in here.